integration/querydsl/blaze-persistence-querydsl-testsuite/src/test/java/com/blazebit/persistence/querydsl/BasicQueryTest.java

```diff
@@ -0,0 +1,744 @@
+                    .select(recursiveEntity)
+                    .from(select(recursiveEntity)
+                            .from(recursiveEntity)
+                            .where(recursiveEntity.parent.name.eq("root1"))
+                            .orderBy(recursiveEntity.name.asc())
+                            .limit(1L), recursiveEntity)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoHibernate42.class, NoHibernate43.class, NoHibernate50.class, NoHibernate51.class})
+    public void testCteInSubquery() {
+        doInJPA(entityManager -> {
+            QIdHolderCTE idHolderCTE = QIdHolderCTE.idHolderCTE;
+
+            List<Document> fetch = new BlazeJPAQuery<>(entityManager, cbf)
+                    .select(document)
+                    .from(document)
+                    .innerJoin(selectFrom(idHolderCTE)
+                                    .with(idHolderCTE, idHolderCTE.id).as(select(document.id).from(document)),
+                            idHolderCTE).on(idHolderCTE.id.eq(document.id))
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    // NOTE: Entity joins are only supported on Hibernate 5.1+
+    // NOTE: Hibernate 5.1 renders t.id = tSub.id rather than t = tSub
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoHibernate42.class, NoHibernate43.class, NoHibernate50.class, NoHibernate51.class})
+    public void testMultipleInlineEntityWithLimitJoin() {
+        doInJPA(entityManager -> {
+            QRecursiveEntity t = new QRecursiveEntity("t");
+            QRecursiveEntity subT = new QRecursiveEntity("subT");
+            QRecursiveEntity subT2 = new QRecursiveEntity("subT2");
+
+            List<RecursiveEntity> fetch = new BlazeJPAQuery<RecursiveEntity>(entityManager, cbf)
+                    .select(t)
+                    .from(new BlazeJPAQuery<RecursiveEntity>().select(t).from(t)
+                            .leftJoin(selectFrom(subT).where(subT.parent.name.eq("root1")).orderBy(subT.name.asc()).limit(1), subT)
+                            .on(t.eq(subT))
+                            .where(t.parent.name.eq("root1"))
+                            .orderBy(t.name.asc())
+                            .limit(1L), t)
+                    .leftJoin(selectFrom(subT2)
+                            .where(subT2.parent.name.eq("root1"))
+                            .orderBy(subT2.name.asc())
+                            .limit(1), subT2)
+                    .on(t.eq(subT2))
+                    .fetch();
+
+            assertNotNull(fetch);
+
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOracle.class, NoOpenJPA.class, NoHibernate42.class, NoHibernate43.class, NoHibernate50.class, NoHibernate51.class, NoH2.class, NoMySQLOld.class })
```

Comment:
Please add comments for the exclusions. The Oracle and old MySQL version we test with doesn't support lateral, H2 also doesn't.